Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -217,11 +218,34 @@ bool WebCookieJar::cookiesEnabled(const Document& document) const
     if (!webFrame || !webFrame->page())
         return false;
 
-    ApplyTrackingPrevention dummy;
-    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), document.cookieURL(), dummy))
+    ApplyTrackingPrevention applyTrackingPreventionInNetworkProcess = ApplyTrackingPrevention::No;
+    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), document.cookieURL(), applyTrackingPreventionInNetworkProcess))
         return false;
 
-    return WebProcess::singleton().ensureNetworkProcessConnection().cookiesEnabled();
+    if (applyTrackingPreventionInNetworkProcess == ApplyTrackingPrevention::No)
+        return true;
+
+    if (!document.cachedCookiesEnabled()) {
+        RELEASE_LOG_ERROR(Network, "WebCookieJar::cookiesEnabled remote is not cached yet");
```

Comment:
In my opinion, this is not any better than the previous iteration. It can still lead to returning an incorrect result.
I think we need to do a sync IPC in this particular edge case, to make sure we consistently return the right value and don't end up with flaky results.